### PR TITLE
Feature/hilbert cli tools

### LINKED ISSUE #13: Configuration Design (super issue)
# Initial design issues:

- [x] Global config (**Q: `Hilbert.yaml`?**) format: mapping not list (#17). **Conclusion**: [general config structure](https://github.com/hilbert/hilbert-cli/issues/17#issuecomment-255210272)
- [x] Data fields should not involve non-standard shortcuts (#20). (examples: `address`, `description`, `services` and `default_application` data fields)
- [x] No use of YAML features like Anchors/Aliases (#16). **Conclusion**: use unique string IDs (see #17)
- [x] Do not use `null` | `void` (non-domain words) (#26).
- [x] Danger of silent overrides due to YAML (#27)
- [x] Station's settings (for Hilbert) (#24). **Conclusion**: https://github.com/hilbert/hilbert-cli/issues/24#issuecomment-257183930
- [x] How to specify resources which are external to the general configuration file (e.g. URI / URL for icons & `docker-compose.yml`) (#18, #9)
- [x] `Service` definition (#19)
- [x] How does `Hilbert-CLI-Server` know from the config which `ServiceRuntime` to use? (#32). Note: same problem with `PowerOnMethod`.
- [x] Structure/definition of a `profile` (#21). **Conclusion**: https://github.com/hilbert/hilbert-cli/issues/21#issuecomment-257062048
- [x] Matching applications with stations (originally: via profiles) (#22) **Obsolete** due to #15.
- [x] n:m relation between stations and applications: **Group** := "set of stations", `Application::compatibleStations`: Group (#15). **Conclusion** is [here](https://github.com/hilbert/hilbert-cli/issues/15#issuecomment-257008776)
- [x] Definition of a Station's remote execution/management methods (Host) (#23). Detailed discussion is in #25. **Conclusion** is https://github.com/hilbert/hilbert-cli/issues/23#issuecomment-257063094.
- [x] Self-contained Access DB in `~/.ssh/config` + station addresses should also be SSH aliases (#25). 
- [x] Rename data fields which are currently in camelCase: `intersectWith`, `compatibleStations`. See https://github.com/hilbert/hilbert-cli/issues/20#issuecomment-266258677

### PR DESCRIPTION
**Pull request by malex984**
Implement / update / fix issues mentioned by #13. Closes #13 
And started fixing issues due to #14.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,11 +15,12 @@ matrix:
       env: TOXENV=py34
 
 install:
-  - ./.travis/install.sh
+  - pip install --upgrade tox
+  - "TOX_ENV=${TRAVIS_PYTHON_VERSION/[0-9].[0-9]/py${TRAVIS_PYTHON_VERSION/.}}"
 
 script:
   - source ~/.venv/bin/activate
```
Comment (porst17) [inline .travis.yml]: Travis CI tests are failing on this line.

**Review by porst17 — CHANGES_REQUESTED** [review]
I enabled automated testing with TavisCI and they are still failing. I know that there are still many tests missing, but the general framework should be in place and working in principle.

**Comment by malex984** [comment]
Seems to be fixed now

**Comment by malex984** [comment]
it seems to be enough for a release candidate now.

**Pull request merged**